src/main/java/com/pingcap/tikv/operation/KVErrorHandler.java

```diff
@@ -53,32 +54,32 @@ public void handle(RespT resp) {
         // no need update here. just let retry take control of this.
         this.regionManager.updateLeader(ctx.getRegionId(), ctx.getPeer().getStoreId());
         // TODO add sleep here
-        throw new StatusRuntimeException(Status.fromCode(Status.Code.UNAVAILABLE));
+        throw new StatusRuntimeException(Status.fromCode(Status.Code.UNAVAILABLE).withDescription("not leader"));
```

Comment:
some error has it's own information. Would you please take a look and channel it through Exception? You might try error.toString() in exception msg.